sds_data_manager/stacks/processing_stack.py

```diff
@@ -94,18 +94,19 @@ def __init__(
             data_bucket=data_bucket,
         )
 
-        # TODO: This will be a construct and also we will add to its capabilities.
-        rule = events.Rule(
-            self,
-            "rule",
-            event_pattern=events.EventPattern(
-                source=["aws.s3"],
-                detail_type=["Object Created"],
-                detail={
-                    "bucket": {"name": [data_bucket.bucket_name]},
-                    "object": {"key": [{"prefix": f"{instrument_sources}"}]},
-                },
-            ),
-        )
+        # Kicks off Step Function as a result of object ingested into directories in
+        # s3 bucket (instrument_sources).
+        for source in instrument_sources:
```

Comment:
Nice, this looks good to me!

Reply:
Is this kicking off a new step function each time one of the dependencies gets a new file? Or is it kicking off an event and a single step function would wait for all of the dependencies to trigger their events?

I'm wondering if we'll have a lot of step functions kicked off and most of them just fail until one of them meets the conditions, or is it more like one step function is kicked off the first time and then has some Waits associated with it that check for new data?

Reply:
Yes. It is a new step function each time. We had a "WAIT" in the original step function design that I can easily add back in. I agree. I think it is the right thing to do.

Reply:
Ok, thought about this more. Right now we are using event-based triggering for the step functions. The dependencies will be listed in dependencies.json so:

"l1b_Codice": ["l1a_Codice", "l1a_Mag"]

l1b codice processing will kick off anytime new objects are dropped in the l1a_Codice or l1a_Mag buckets. The lambda within the step function will check to see if requirements for processing are met and end (not fail) the step function if not. If we add a "Wait" in the step function there will still be step functions that are triggered since we are using event-based processing.